### ISSUE TITLE
[PPC] VM address overwritten to 0 by aliased mnemonics

### ISSUE THREAD

**Issue by EdwardLarson** [keystone-engine/keystone#427]
Assembling certain instructions will reset the vm address of the instructions being assembled to 0. When assembling a sequence of instructions e.g. ("add 3, 1 ,2; add 3, 3 4"), this causes all relative offset branches following the problematic instruction to encode an incorrect target.

For example, the following writes a branch instruction to itself:
```
kstool ppc32be "b 0x104" 0x104
b 0x104 = [ 48 00 00 00 ]
```
You would expect the same machine code to be emitted when assembling multiple instructions such that the branch instruction is at the same address:
```
kstool ppc32be "nop; b 0x104" 0x100
nop; b 0x104 = [ 60 00 00 00 48 00 00 00 ]
```
However `slwi` is one of the problematic instructions which resets the address to 0, causing incorrect machine code to be emitted:
```
kstool ppc32be "slwi 0, 0, 0; b 0x104" 0x100
slwi 0, 0, 0; b 0x104 = [ 54 00 00 3e 48 00 01 00 ]
```
You can see the address of the branch instruction, which should be 0x104, is actually 0x004, causing an offset of 0x100 to be encoded.

**Comment by EdwardLarson** [comment]
The issue is in PPCAsmParser::ProcessInstruction in /llvm/lib/Target/PowerPC/AsmParser/PPCAsmParser.cpp. All instructions which have a case in the switch statement will cause this address reset. The fix is simple and I will submit a patch.

**Comment by EdwardLarson** [comment]
Another long-fixed issue (by #429)
